pkg/gui/gui.go

```diff
@@ -363,5 +363,8 @@ func (gui *Gui) RunWithSubprocesses() {
 }
 
 func (gui *Gui) quit(g *gocui.Gui, v *gocui.View) error {
-	return gocui.ErrQuit
+	if v.Name() != "commitMessage" && v.Name() != "confirmation" {
```

Comment:
Could you also add "main": we can escape from main when we're resolving merge conflicts. Also we should rename this function to 'escape' now that it's generally just a function for when you either want to escape the current view, or escape the app.

I'd prefer a solution where globally set keybindings are ignored if there is a matching keybinding for the current view; that way we wouldn't need to add to this list for every new view that is escapable.

What are your thoughts of patching the gocui fork like so. It restricts keybindings to only be matched with one handler at a time, with preference given to matching on views over matching globally.

```
diff --git a/vendor/github.com/jesseduffield/gocui/gui.go b/vendor/github.com/jesseduffield/gocui/gui.go
index 28a52d1..b746ca0 100644
--- a/vendor/github.com/jesseduffield/gocui/gui.go
+++ b/vendor/github.com/jesseduffield/gocui/gui.go
@@ -653,17 +653,29 @@ func (g *Gui) onKey(ev *termbox.Event) error {
 // execKeybindings executes the keybinding handlers that match the passed view
 // and event. The value of matched is true if there is a match and no errors.
 func (g *Gui) execKeybindings(v *View, ev *termbox.Event) (matched bool, err error) {
-       matched = false
+       var globalKb *keybinding
        for _, kb := range g.keybindings {
                if kb.handler == nil {
                        continue
                }
-               if kb.matchKeypress(Key(ev.Key), ev.Ch, Modifier(ev.Mod)) && kb.matchView(v) {
+               if !kb.matchKeypress(Key(ev.Key), ev.Ch, Modifier(ev.Mod)) {
+                       continue
+               }
+               if kb.matchView(v) {
                        if err := kb.handler(g, v); err != nil {
                                return false, err
                        }
-                       matched = true
+                       return true, nil
+               }
+               if kb.viewName == "" {
+                       globalKb = kb
+               }
+       }
+       if globalKb != nil {
+               if err := globalKb.handler(g, v); err != nil {
+                       return false, err
                }
+               return true, nil
        }
-       return matched, nil
+       return false, nil
 }
diff --git a/vendor/github.com/jesseduffield/gocui/keybinding.go b/vendor/github.com/jesseduffield/gocui/keybinding.go
index 7efdb75..82d1acc 100644
--- a/vendor/github.com/jesseduffield/gocui/keybinding.go
+++ b/vendor/github.com/jesseduffield/gocui/keybinding.go
@@ -38,9 +38,6 @@ func (kb *keybinding) matchView(v *View) bool {
        if v.Editable == true && kb.ch != 0 {
                return false
        }
-       if kb.viewName == "" {
-               return true
-       }
        return v != nil && kb.viewName == v.name
 }
```

Reply:
Cool works perfectly, commit that to your fork and I'll update the dep.